command/operator/autopilot/state/formatter.go

```diff
@@ -0,0 +1,157 @@
+package state
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/consul/api"
+)
+
+const (
+	PrettyFormat string = "pretty"
+	JSONFormat   string = "json"
+)
+
+// Formatter defines methods provided by an autopilot state output formatter
+type Formatter interface {
+	FormatState(state *api.AutopilotState) (string, error)
+}
+
+// GetSupportedFormats returns supported formats
+func GetSupportedFormats() []string {
+	return []string{PrettyFormat, JSONFormat}
+}
+
+// NewFormatter returns Formatter implementation
+func NewFormatter(format string) (formatter Formatter, err error) {
+	switch format {
+	case PrettyFormat:
+		formatter = newPrettyFormatter()
+	case JSONFormat:
+		formatter = newJSONFormatter()
+	default:
+		err = fmt.Errorf("Unknown format: %s", format)
+	}
+
+	return formatter, err
+}
+
+func newPrettyFormatter() Formatter {
+	return &prettyFormatter{}
+}
+
+type prettyFormatter struct {
+}
+
+func outputStringSlice(buffer *bytes.Buffer, indent string, values []string) {
+	for _, val := range values {
+		buffer.WriteString(fmt.Sprintf("%s%s\n", indent, val))
+	}
+}
+
+func (f *prettyFormatter) FormatState(state *api.AutopilotState) (string, error) {
+	var buffer bytes.Buffer
+
+	buffer.WriteString(fmt.Sprintf("Healthy:                      %t\n", state.Healthy))
+	buffer.WriteString(fmt.Sprintf("Failure Tolerance:            %d\n", state.FailureTolerance))
+	buffer.WriteString(fmt.Sprintf("Optimistic Failure Tolerance: %d\n", state.OptimisitcFailureTolerance))
```

Comment:
I realized this when looking at the output, we've misspelled `Optimistic` in the api models as `Optimisitc`!

Reply:
Yep, I saw that too a few minutes ago and am fixing it.